seedlink_plotter/seedlink_plotter.py

```diff
@@ -474,9 +497,61 @@ def get_events(self):
         with self.lock:
             start = min([tr.stats.starttime for tr in self.stream])
             end = max([tr.stats.endtime for tr in self.stream])
-        neries_emsc = Client("EMSC")
-        events = neries_emsc.get_events(starttime=start, endtime=end,
-                                          minmagnitude=self.args.events)
+        
+        events = Catalog()
+        
+         # get the coordinats of the station
+        splitted_ids = [x.split(".") for x in self.args.ids]  # use seedlink id to get network, station, location and channel
+        for client in ["iris-federator","eida-routing"]:
+            try:
+                Cli = RoutingClient(client)
+                station_inventory = Cli.get_stations(network = splitted_ids[0][0],
+                                   station = splitted_ids[0][1],
+                                   channel = splitted_ids[0][3],
+                                   location = splitted_ids[0][2])
+                
+                station_latidude = station_inventory[0][0].latitude 
+                station_longitude = station_inventory[0][0].longitude 
+                break
+            except:
+               pass
+        
+        
+        # search for events fullfilling each criteria
+        for criteria in self.args.events_criteria:
+            print("Search for events with criteria" + str(criteria))
+            temp_catalog = Catalog()
+            for client in URL_MAPPINGS.keys():
+                try:
+                    cli = Client(client)
+                    
+                    # get all events within a rdius (in deg) around the station with a minimum magnitude
+                    t = cli.get_events(starttime=start, endtime=end,
+                                                minmagnitude=criteria[0],
+                                                maxmagnitude=criteria[1],                                          
+                                                latitude = station_latidude,
+                                                longitude = station_longitude,
+                                                minradius = criteria[2],
+                                                maxradius = criteria[3])
+                    if t.count() > temp_catalog.count():
+                        temp_catalog = t
+                except:
+                    pass
+                
+            events.extend(temp_catalog)
+            if temp_catalog.count == 0:
+                print("No events found for the current settings " +str(criteria) + "\n")
+            else:
+            	print(str(temp_catalog.count()) + " events found" )
+
+        # corrects the time to be the arrival time and not the local event time
+        if events.count() > 0:
+            try:
+                events = self.event_time_to_arrival_time(events, station_latidude, station_longitude)
```

Comment:
Fetching the station metadata should only happen once after init. Station metadata should then be attached to `self` and be used from there inside `self.event_time_to_arrival_time`. Also the initialization of clients used to fetch events should only happen once during init and be stored with `self`